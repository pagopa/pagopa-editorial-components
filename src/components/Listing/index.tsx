import { Box, Divider, Typography, Link } from "@mui/material";
import EContainer from "../EContainer";

interface ListingItem {
  text: string,
  href?: string,
  htmlTitle?: string,
  isLast?: boolean
}

export interface ListingsProps {
  title: string,
  date: {
    preDate?: string,
    date: Date,
    locale?: string,
    options: Intl.DateTimeFormatOptions,
  }
  name: string,
  items: ListingItem[]
}

const ListingItem = (props: ListingItem) => {
  const { text, htmlTitle = '', href, isLast = false } = props;
  return (
    <Box mb={ isLast ? 0 : 2}>
      {
        href ? (
          <Link
            href={href}
            target="_blank"
            mb={2}
            variant="body2"
            fontWeight={600}
            title={htmlTitle}>
            {text}
          </Link>
        ) : (
          <Typography
            fontWeight={600}
            mb={2}
            variant="body2"
            title={htmlTitle}>
            {text}
          </Typography>
        )
      }
      <Divider sx={{ marginTop:  '16px' }}/>
    </Box>
  )
}

const Listing = (props: ListingsProps) => {
    const { 
      title,
      date: {
        preDate,
        date,
        locale,
        options
      },
      name = '', items
    } = props;
    return (
      <EContainer py={4} background='background.paper' direction='column'>
        <Typography variant='h5' mb={2}>
          {title}
        </Typography>
        <Typography
          color='text.secondary'
          fontSize={12}
          fontWeight={600}
          mb={5}>
          {
            `${preDate} ${new Intl
              .DateTimeFormat(locale, options)
              .format(date)}`
          }
        </Typography>
        <Typography
          color='text.secondary'
          fontSize={14}
          fontWeight={700}
          mb={2}
          textTransform='uppercase'>
            {name}
          </Typography>
        {
          items.map( (props, i) =>
            <ListingItem
              key={`${props.text}-${i}`}
              { ...props }
              isLast={items.length === i+1 }/>)
        }
      </EContainer>
    )
}

export default Listing;
