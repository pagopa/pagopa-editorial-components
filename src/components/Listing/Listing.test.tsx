import Listing from './index';
import '@testing-library/jest-dom';
/**
 * @jest-environment jsdom
 */

import renderer from 'react-test-renderer';

it('renders correctly standard ', () => {
  const tree = renderer
    .create(
      <Listing
        date={{
          date: new Date('2023-11-23T11:22:53.167Z'),
          locale: 'it-IT',
          options: {
            day: '2-digit',
            month: 'long',
            year: 'numeric',
          },
          preDate: 'Ultimo aggiornamento',
        }}
        items={[
          {
            href: 'https://google.it',
            htmlTitle: 'vai al sito',
            text: 'Lorem ipsum dolor sit amet, consectetuer asipscing elit',
          },
          {
            text: 'Lorem ipsum dolor sit amet, consectetuer asipscing elit',
          },
          {
            text: 'Lorem ipsum dolor sit amet, consectetuer asipscing elit',
          },
          {
            text: 'Lorem ipsum dolor sit amet, consectetuer asipscing elit',
          },
        ]}
        name="Nome"
        title="Lorem ipsum"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
