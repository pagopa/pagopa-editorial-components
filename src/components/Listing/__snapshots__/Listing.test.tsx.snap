// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly standard  1`] = `
<section
  className="MuiBox-root css-1cct0at"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-1p4tas-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 css-ch2k2f-MuiTypography-root"
      >
        Lorem ipsum
      </h5>
      <p
        className="MuiTypography-root MuiTypography-body1 css-bsld86-MuiTypography-root"
      >
        Ultimo aggiornamento 23 novembre 2023
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-12akxqs-MuiTypography-root"
      >
        Nome
      </p>
      <div
        className="MuiBox-root css-1qm1lh"
      >
        <a
          className="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-v8rapa-MuiTypography-root-MuiLink-root"
          href="https://google.it"
          onBlur={[Function]}
          onFocus={[Function]}
          target="_blank"
          title="vai al sito"
        >
          Lorem ipsum dolor sit amet, consectetuer asipscing elit
        </a>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-uok36i-MuiDivider-root"
        />
      </div>
      <div
        className="MuiBox-root css-1qm1lh"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 css-kah6a3-MuiTypography-root"
          title=""
        >
          Lorem ipsum dolor sit amet, consectetuer asipscing elit
        </p>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-uok36i-MuiDivider-root"
        />
      </div>
      <div
        className="MuiBox-root css-1qm1lh"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 css-kah6a3-MuiTypography-root"
          title=""
        >
          Lorem ipsum dolor sit amet, consectetuer asipscing elit
        </p>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-uok36i-MuiDivider-root"
        />
      </div>
      <div
        className="MuiBox-root css-12595yo"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 css-kah6a3-MuiTypography-root"
          title=""
        >
          Lorem ipsum dolor sit amet, consectetuer asipscing elit
        </p>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-uok36i-MuiDivider-root"
        />
      </div>
    </div>
  </div>
</section>
`;
